vulcanai2/models/basenetwork.py

```diff
@@ -281,6 +285,19 @@ def _create_network(self):
         :return: None
         """
         pass
+    
+    def forward(self, *inputs):
+        """Perform a forward pass through the module/modules."""
+
+        if isinstance(self.input_network, list):
+            outputs = []
+            for module in self.input_network:
+                outputs.append(module._forward(inputs))
```

Comment:
You are right! Like I mentioned this is not a working program yet I have to fix the code but before that I wanted to have a discussion if this is how @rfratila wanted.

Reply:
yeah we're on the right track. While we should have some sort of "get forward through all input networks" type of function, we can have it as a sister function to `forward` (which is specific to each module) in base network instead of overriding forward from basenetwork. Also I've been thinking more about the self.input_networks object and it should be a dict so we can attach options like freezing the weights of specific inputs @priyatharsan

Reply:
regarding @nirtiac 's comment, that would be one way to do it. The most important thing to keep in mind is making sure the right data inputs goes into the correct module.

Reply:
we will have to be very careful with the `outputs` list since the outputs from networks may not immediately compatible for concatenation. say we have a Conv output of shape `[16, 64, 28, 28]` and a dense output of shape `[16, 400]`, in this scenario we should probably linearize the conv output if it is feeding into a dense net (like the way you currently have it). Something else to think about is if you wanted a Conv + dense to feed into conv, do you maintain the conv output of the input network and just append the dense outputs to the shape of conv? does that make sense? im curious what you are thinking @nirtiac @priyatharsan